commit a28ce49c8373a261aad04ef37fdf1d364c56541e
Author: Isaac Davis <isaac.davis@joyent.com>
Date:   2019-03-26T01:02:55+00:00 (7 months ago)
    
    TRITON-1351 Add CLI flag for ssh key in LX Prometheus and Grafana setup scripts

diff --git a/setup-prometheus-prod.sh b/setup-prometheus-prod.sh
index 50b24ba..e7baa9a 100755
--- a/setup-prometheus-prod.sh
+++ b/setup-prometheus-prod.sh
@@ -23,7 +23,7 @@ PROMETHEUS_VERSION="2.3.2"
 ALIAS=prometheus0
 
 function usage {
-    echo "usage: ./setup-prometheus.sh [-i] [-f] [-r '<extra resolver 1>,<extra resolver 2>'] [-s <non-local server uuid>]" >&2
+    echo "usage: ./setup-prometheus.sh [-i] [-f] [-r '<extra resolver 1>,<extra resolver 2>'] [-s <non-local server uuid>] [-k <path to ssh key>]" >&2
     exit 1
 }
 
@@ -74,9 +74,11 @@ function get_admin_ip() {
 
 insecure_flag="false"
 firewall_flag="false"
+ssh_key_file=
+ssh_key=
 resolvers=
 server_uuid=$(sysinfo | json UUID) # local headnode by default
-while getopts ":ifr:s:" f; do
+while getopts ":ifr:s:k:" f; do
     case $f in
         i)  insecure_flag="true"
             ;;
@@ -86,12 +88,14 @@ while getopts ":ifr:s:" f; do
             ;;
         s)  server_uuid=$OPTARG
             ;;
+        k)  ssh_key_file=$OPTARG
+            ;;
         \?) usage
             ;;
     esac
 done
 
-if [[ $# -gt 6 ]]; then
+if [[ $# -gt 8 ]]; then
     usage
 fi
 
@@ -108,6 +112,12 @@ fi
 
 . ~/.bash_profile
 
+# Check that key exists if we passed the flag; then read the key's contents
+if [[ -n "${ssh_key_file}" ]]; then
+    [[ -f "${ssh_key_file}" ]] || fatal "ssh key not found at ${ssh_key_file}"
+    ssh_key=$(<"${ssh_key_file}")
+fi
+
 #
 # prometheus0 zone creation
 #
@@ -277,6 +287,11 @@ cat > /zones/${vm_uuid}/root/etc/systemd/system/prometheus.service <<SYSTEMD
     WantedBy=multi-user.target
 SYSTEMD
 
+# Add ssh key, if specified
+if [[ -n "${ssh_key_file}" ]]; then
+    echo "${ssh_key}" >> /zones/${vm_uuid}/root/root/.ssh/authorized_keys
+fi
+
 zlogin ${vm_uuid} "systemctl daemon-reload && systemctl enable prometheus && systemctl start prometheus && systemctl status prometheus" < /dev/null
 SERVER
 
